### PR TITLE
tests: add github action integration

### PR DESCRIPTION
**Pull request by flotter**
Enable automated Go checks on pull requests and merge requests.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+jobs:
+  build-test:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ['1.18', '1.17', '1.16', '1.15', '1.14', '1.13']
+
+    name: basic-with-go-v${{ matrix.go }}"
```
Comment (benhoyt) [inline .github/workflows/tests.yaml:12]: Looks like this is missing a leading `"`.

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    name: "static-check-linter-2022.1"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - uses: dominikh/staticcheck-action@v1.2.0
+        with:
+          version: "2022.1"
```
Comment (benhoyt) [inline .github/workflows/tests.yaml:51]: Nit: would be nice to have a blank line before `coverage:`.

File: i18n/i18n.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
```
Comment (benhoyt) [inline i18n/i18n.go:1]: Can we move these other `.go` changes out of this PR? As it stands, the PR also includes all the i18n stuff so is hard to review in isolation.

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  static-check:
+    name: "static-check-linter-2022.1"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - uses: dominikh/staticcheck-action@v1.2.0
```
Comment (mvo5) [inline .github/workflows/tests.yaml:49]: Out of curiosity, what static checks are included in this action?

File: i18n/i18n.go (hunk 3 above)
Comment (mvo5) [reply i18n/i18n.go:1]: Fwiw, I agree

**Review by mvo5 — COMMENTED** [review]
A bit of a meta-comment/nitpick: For snapd there is a convention to have the PR summary follow the form: "<affected-area-or-module>: <short-summary>". E.g. `strutil: fix foobar` - I wonder if there is a desire to follow similar patterns here?

File: i18n/i18n.go (hunk 3 above)
Comment (flotter) [reply i18n/i18n.go:1]: Sorry the Go changes will disappear the moment I merge the first PR in the chain, and rebase this one on main. Sorry for the confusion. This is just a side effect of working on chain of PRs and Github does not allow a delta PR view (minus dependencies on previous PRs)

**Review by benhoyt — APPROVED** [review]

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (flotter) [reply .github/workflows/tests.yaml:12]: Removed the double quote. Thanks.

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (flotter) [reply .github/workflows/tests.yaml:51]: Fixed.

File: i18n/i18n.go (hunk 3 above)
Comment (flotter) [reply i18n/i18n.go:1]: Removed after rebase.

File: .github/workflows/tests.yaml (hunk 4 above)
Comment (flotter) [reply .github/workflows/tests.yaml:49]: https://staticcheck.io/

Discussed in 1-1 meeting.

**Comment by flotter** [comment]
> A bit of a meta-comment/nitpick: For snapd there is a convention to have the PR summary follow the form: ": ". E.g. `strutil: fix foobar` - I wonder if there is a desire to follow similar patterns here?

I like it. Will follow that for future PRs.

**Pull request merged**